File: 01-week-1--starter-algorithms/03-day-4--selection-sort/ruby/selection_sort.rb
def selection_sort(arr)

  n = arr.length - 1
 
    n.times do |i|
      min_index = i
      for j in (i + 1)..n 
        min_index = j if arr[j] < arr[min_index]
      end
      arr[i], arr[min_index] = arr[min_index], arr[i]
    end
  return arr
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [-1, 2, 3, 5]"
  print "=> "
  print selection_sort([3, -1, 5, 2])

  puts

  puts "Expecting: [-8, -4 , 3, 12]"
  print "=> "
  print selection_sort([12, -4, 3, -8])

  puts

  # Don't forget to add your own!

  # BENCHMARK HERE, you can print the average runtime
  long_input = []

  100.times { long_input << rand }
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 01-week-1--starter-algorithms/03-day-4--selection-sort/javascript/selection_sort.js
function selectionSort(arr) {
  // type your code here
  for(let i = 0; i < arr.length - 1; i++){
      let minIndex = i
      for(let j = i + 1; j < arr.length; j++){
        if(arr[j] < arr[minIndex]){
          minIndex = j
        }
      }
      [arr[i], arr[minIndex]] =[arr[minIndex], arr[i]]
  }
  return arr
}

//You have to create a for loop where it would iterate through an array for a set number of times
//(depending on the length of the array) to then first find the smallest number. Once it finds the smallest
//number, the loop would set it as the minimum number and place said number in the front of the array.
//The process would continue until all numbers have been sorted 

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: [-1, 2, 3, 5]");
  console.log("=>", selectionSort([3, -1, 5, 2]));

  console.log("");

  // BENCHMARK HERE, and print the average runtime
  const longInput = [];

  for (let i = 0; i < 100; ++i) {
    longInput.push(Math.random());
  }
}

module.exports = selectionSort;

// Please add your pseudocode to this file
// And a written explanation of your solution
